pygdf/cudautils.py

```diff
@@ -266,6 +267,40 @@ def compact_mask_bytes(boolbytes):
     return bits
 
 
+@cuda.jit
+def gpu_mask_from_devary(ary, bits):
+    tid = cuda.grid(1)
+    base = tid * mask_bitsize
+    for i in range(base, base + mask_bitsize):
+        if i >= len(ary):
+            break
+        if not isnan(ary[i]):
+            mask_set(bits, i)
+
+
+@cuda.jit
+def gpu_mask_from_devary_int(ary, bits):
+    tid = cuda.grid(1)
+    base = tid * mask_bitsize
+    for i in range(base, base + mask_bitsize):
+        if i >= len(ary):
+            break
+        if not isnan(float(ary[i])):
```

Comment:
Casting every integer type to a float seems wasteful but not sure if we have other options, @dantegd any ideas?

Reply:
So the definition of nan, in particular that `nan != nan` was staring me in the face, so I changed this to 

```
if ary[i] == ary[i]:
            mask_set(bits, i)
```
which removes the need to check for int or float. @scopatz @kkraus14 does this sound correct or is there a reason not to do that that I'm missing?

Reply:
I am not super familar with how the JIT is going to transform the code here, but it seems like things like the `isnan()` should only be run in the event of a float, so for other types it doesn't make sense to convert to float.  That could yield some weird errors. Also, for some data types this conversion could be pretty expensive.